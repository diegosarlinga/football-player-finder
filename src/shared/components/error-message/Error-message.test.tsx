import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import ErrorMessage from './Error-message';

describe('ErrorMessage component tests', () => {
  let wrapper: ShallowWrapper;

  describe('when rendered with no props', () => {
    beforeEach(() => {
      wrapper = shallow(<ErrorMessage/>);
    });
  
    it('should match default snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

  });

  describe('when rendered with props', () => {
    beforeEach(() => {
      const props = {
        title: 'Error title',
        text: 'Ups, something went really wrong'
      };
      wrapper = shallow(<ErrorMessage {...props}/>);
    });
  
    it('should match custom text snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

  });

});
