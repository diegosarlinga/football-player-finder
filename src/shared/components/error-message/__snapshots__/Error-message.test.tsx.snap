// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component tests when rendered with no props should match default snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-body d-flex flex-row justify-content-center align-items-center"
  >
    <ForwardRef(SvgErrorIcon)
      className="Error-icon"
    />
    <div
      className="d-flex flex-column align-items-start"
    >
      <h5
        className="card-title"
      >
        Error
      </h5>
      <p
        className="card-text"
      >
        Something went wrong
      </p>
    </div>
  </div>
</div>
`;

exports[`ErrorMessage component tests when rendered with props should match custom text snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-body d-flex flex-row justify-content-center align-items-center"
  >
    <ForwardRef(SvgErrorIcon)
      className="Error-icon"
    />
    <div
      className="d-flex flex-column align-items-start"
    >
      <h5
        className="card-title"
      >
        Error title
      </h5>
      <p
        className="card-text"
      >
        Ups, something went really wrong
      </p>
    </div>
  </div>
</div>
`;
